mindocr/data/transforms/det_fce_transforms.py

```diff
@@ -0,0 +1,1461 @@
+                if h < w:
+                    ratio = float(limit_side_len) / h
+                else:
+                    ratio = float(limit_side_len) / w
+            else:
+                ratio = 1.0
+        elif self.limit_type == "resize_long":
+            ratio = float(limit_side_len) / max(h, w)
+        else:
+            raise Exception("not support limit type, image ")
+        resize_h = int(h * ratio)
+        resize_w = int(w * ratio)
+
+        resize_h = max(int(round(resize_h / 32) * 32), 32)
+        resize_w = max(int(round(resize_w / 32) * 32), 32)
+
+        if int(resize_w) <= 0 or int(resize_h) <= 0:
+            return None, (None, None)
+        img = cv2.resize(img, (int(resize_w), int(resize_h)))
+        ratio_h = resize_h / float(h)
+        ratio_w = resize_w / float(w)
+        return img, [ratio_h, ratio_w]
+
+    def resize_image_type2(self, img):
+        h, w, _ = img.shape
+
+        resize_w = w
+        resize_h = h
+
+        if resize_h > resize_w:
+            ratio = float(self.resize_long) / resize_h
+        else:
+            ratio = float(self.resize_long) / resize_w
+
+        resize_h = int(resize_h * ratio)
+        resize_w = int(resize_w * ratio)
+
+        max_stride = 128
+        resize_h = (resize_h + max_stride - 1) // max_stride * max_stride
+        resize_w = (resize_w + max_stride - 1) // max_stride * max_stride
+        img = cv2.resize(img, (int(resize_w), int(resize_h)))
+        ratio_h = resize_h / float(h)
+        ratio_w = resize_w / float(w)
+
+        return img, [ratio_h, ratio_w]
+
+
+class FDetLabelEncode(object):
+    def __init__(self, min_area=100, **kwargs):
+        self.min_area = min_area
+        # pass
+
+    def __call__(self, data):
+        data["image"] = cv2.imread(data["img_path"], cv2.IMREAD_COLOR)[:, :, ::-1]
+        label = data["label"]
+        label = json.loads(label)
+        nBox = len(label)
+        boxes, txts, txt_tags = [], [], []
+        for bno in range(0, nBox):
+            box = label[bno]["points"]
+
+            if np.sum(np.array(box, dtype=np.float32), axis=0).min() < 1:
+                continue
+            txt = label[bno]["transcription"]
+            boxes.append(box)
+            txts.append(txt)
+            if txt in ["*", "###"]:
+                txt_tags.append(True)
+            else:
+                txt_tags.append(False)
+        if len(boxes) == 0:
+            return None
+        boxes = self.expand_points_num(boxes)
+        # boxes = np.array(boxes, dtype=np.float32)
+        # txt_tags = np.array(txt_tags, dtype=np.bool)
+        data["polys"] = np.array(boxes, dtype=np.float32)
+        data["texts"] = txts
+        data["ignore_tags"] = txt_tags
+        data["polygons"] = np.array([boxes[i] for i in range(len(txt_tags)) if not txt_tags[i]], dtype=np.float32)
+        data["ignore_polygons"] = np.array([boxes[i] for i in range(len(txt_tags)) if txt_tags[i]], dtype=np.float32)
```

Comment:
Why do you split polygons to `polygons` and `ignore_polygons` if you have `ignore_tags`? Anyway, during the augmentation stage need to apply transformations to all polygons, isn't it?